### DISCUSSION TITLE
Using multiple -draw decoration (UnderLine + OverLine) on a single text

### DISCUSSION THREAD

**Discussion by schrmh** [ImageMagick/ImageMagick#6235] [Help]
### ImageMagick version

ImageMagick 7.1.0-52 beta Q32-HDRI x86_64 aea87b538:20221016 https://imagemagick.org

### Operating system, version and so on

Arch Linux x86_64

### Description

I asked this 3 1/2 years ago on the old forum but it did not receive any answer: https://legacy.imagemagick.org/discourse-server/viewtopic.php?f=1&t=36551

To keep it simple, I want to use **UnderLine** and **OverLine** together to show lines above and below a text without using **text** twice.
Use case is to simplify this shell script:
```sh
#!/bin/sh

quote="$1"
author="$2"
date="$3"
  
  convert -background white -size 638x821 \
          -fill firebrick3 -font Arial -gravity North -pointsize 84 \
          caption:"»$quote«" \
          -background none -fill black -font Arial -gravity Center -pointsize 32 \
          caption:"$author" \
          -draw  "gravity South font Arial  font-size 72  decorate OverLine \
          fill dodgerblue \
          translate 0,250 rotate -7 text 0,0 '      $date      ' \
          decorate UnderLine \
          text 0,0 '      $date      '" \
          -flatten \
          wrongquote.png
```
(it generates images inspired by "Der falsche Kalender 2" which translates to "the wrong calendar 2": https://literatour.blog/wp-content/uploads/sites/11/2019/01/IMG_9425.jpg)

**Comment by snibgo** [comment]
In `-draw`, each `text` can have only one `decorate`. If you want `decorate` twice, you need to have `text` twice.

You could create a similar effect by writing text to an image, and drawing horizontal lines at the top and bottom, and compositing that over the main image.
